### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by savchuksavchuk**
[DEMO LINK](https://savchuksavchuk.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/components/TodoItem.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+};
+
+export const TodoItem: React.FC<Props> = React.memo(
+  ({
+    todo,
+    isProcessing,
+    onDelete,
+    onRename,
+    onComplete,
+  }) => {
```
Comment (polosanya) [inline src/components/TodoItem.tsx:20]:
```suggestion
export const TodoItem: React.FC<Props> = React.memo(({
    todo,
    isProcessing,
    onDelete,
    onRename,
    onComplete,
  }) => {
```

File: src/components/TodoItem.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    const [isRenaming, setIsRenaming] = useState(false);
+    const [title, setTitle] = useState(todo.title);
+    const [newTitle, setNewTitle] = useState(todo.title);
+    const renameInput = useRef<HTMLInputElement>(null);
+
+    const onRenameHander = async (event: React.FormEvent<HTMLFormElement>) => {
```
Comment (polosanya) [inline src/components/TodoItem.tsx:26]:
```suggestion
    const handleRename = async (event: React.FormEvent<HTMLFormElement>) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

File: src/components/TodoList.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  isAdding: boolean,
+  newTodoTitle: string,
+  onDelete: (todoId: number) => void,
+  onRename: (todoId: number, title: string) => void,
+  isProcessing: number[],
+  onComplete: (todo: Todo) => void,
+};
+
+export const TodoList: React.FC<Props> = React.memo(
+  ({
```
Comment (polosanya) [inline src/components/TodoList.tsx:18]:
```suggestion
export const TodoList: React.FC<Props> = React.memo(({
```

File: src/components/TodoList.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        <TransitionGroup component={null}>
+          {visibleTodos.map(todo => (
+            <CSSTransition
+              key={todo.id}
+              classNames="item"
+              timeout={300}
+            >
+              <TodoItem
+                key={todo.id}
+                todo={todo}
```
Comment (polosanya) [inline src/components/TodoList.tsx:37]:
```suggestion
              <TodoItem
                todo={todo}
```
No need to add key for TodoItem as you've added it to its parent tag

**Review by polosanya — CHANGES_REQUESTED** [review]
Almost perfect! Let's improve your code a bit)

**Review by Bohdan-Maksymiuk — APPROVED** [review]
The same comments as in previous work
fix them there

**Pull request closed without merging**